src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Insights/New-ArmApplicationInsightsAlertRule.ps1

```diff
@@ -0,0 +1,79 @@
+function New-ArmApplicationInsightsAlertRule {
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    [OutputType("ApplicationInsightsAlertRule")]
+    Param(
+        [Parameter(Mandatory, ValueFromPipeline)]
+        [ValidatePattern('^(\[.*\]|[a-zA-Z0-9-]*)$')]
+        [string]
+        $Name,
+        [Parameter(Mandatory)]
+        [PSTypeName("ApplicationInsights")]
+        $ApplicationInsights,
+        [string]
+        $ApiVersion = '2016-03-01',
+        [string]
+        $Description = "",
+        [switch]
+        $Disable,
+        [ValidateSet(
+            "GreaterThan",
+            "GreaterThanOrEqual",
+            "LessThan",
+            "LessThanOrEqual")]
+        [string]
+        $Operator,
+        [int]
+        $Threshold,
+        [ValidateSet(
+            "Average",
+            "Minimum",
+            "Maximum",
+            "Total",
+            "Last")]
+        [string]
+        $TimeAggregation,
+        [int]
+        $FailedLocationCount,
+        [string]
+        $WindowSize = "PT5M",
+        [ValidateSet(
+            "RuleCondition",
+            "Microsoft.Azure.Management.Insights.Models.ThresholdRuleCondition",
+            "Microsoft.Azure.Management.Insights.Models.LocationThresholdRuleCondition",
+            "Microsoft.Azure.Management.Insights.Models.ManagementEventRuleCondition")]
+        [string]
+        $ConditionOdataType,
```

Comment:
We should rename this to Condition and have a simplified ValidSet and then in the function map to the correct long OdataType